manifests/init.pp

```diff
@@ -1,10 +1,17 @@
-# @summary A short summary of the purpose of this class
+# @summary The main class for buildkite_agent
 #
-# A description of what this class does
+# Entrypoint for the module
 #
 # @example
 #   include buildkite_agent
-class buildkite_agent {
+class buildkite_agent (
+  Optional[Hash[String, Hash[String, Variant[String, Integer, Boolean]]]] $configs,
```

Comment:
@call Can you advise why these config items are nested as such?

Reply:
Hey @dquills That's a very precise [data type definition](https://puppet.com/docs/puppet/5.5/lang_data_type.html) in Puppet. The `buildkite_agent::configs` data structure must be a hash of hashes, and keys can be `str`, `bool`, or `int`. You can find an example of this data structure in the sample Hiera data, specifically this part:

```
buildkite_agent::configs:
  my-bk-config:
    config_file_path: '/usr/local/etc/buildkite-agent/buildkite-agent.cfg'
    bk_name: 'foofoobar'
    tags: 'queue=foobarbar'
    debug: true
    priority: 1
```

Multiple Buildkite agents can run on one machine, so you can use yaml data like the sample below to manage multiple config files:

```
buildkite_agent::configs:
  my-bk-config:
    config_file_path: '/usr/local/etc/buildkite-agent/buildkite-agent.cfg'
    bk_name: 'foofoobar'
    tags: 'queue=foobarbar'
    debug: true
    priority: 1
  my-bk-config:
    config_file_path: '/Users/call/.buildkite-agent/buildkite-agent.cfg'
    bk_name: 'myagent'
    tags: 'queue=myqueue'
    debug: false
    priority: 2
```

Reply:
Ah, that makes sense. Thanks for the explanation.